lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -0,0 +1,39 @@
+const moment = require('moment');
+
+/**
+ * isObjectLocked - checks whether object is locked or not
+ * @param {obect} bucket - bucket metadata
+ * @param {object} objectMD - object metadata
+ * @param {array} headers - request headers
+ * @return {boolean} - indicates whether object is locked or not
+ */
+function isObjectLocked(bucket, objectMD, headers) {
+    if (!bucket.isObjectLockEnabled()) {
+        return false;
+    }
+    const objectLegalHold = objectMD.legalHold;
+    if (objectLegalHold) {
+        return true;
+    }
+    const objectRetention = objectMD.retentionInfo ?
+        objectMD.retentionInfo.retention : '';
```

Comment:
I think this can be converted to:
```suggestion
    const objectRetention = objectMD.retentionInfo && objectMD.retentionInfo.retention;
```

IIRC && will return the right side if the left is true otherwise it will return undefined which is valid for the next if.
![image](https://user-images.githubusercontent.com/8585244/84682459-84b11700-aeea-11ea-8000-391f718e5864.png)

Reply:
neat